x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/LogsDBPlugin.java

```diff
@@ -17,23 +17,31 @@
 import java.util.Collection;
 import java.util.List;
 
+import static org.elasticsearch.xpack.cluster.settings.ClusterSettings.CLUSTER_LOGSDB_ENABLED;
 import static org.elasticsearch.xpack.logsdb.SyntheticSourceLicenseService.FALLBACK_SETTING;
 
 public class LogsDBPlugin extends Plugin {
 
     private final Settings settings;
     private final SyntheticSourceLicenseService licenseService;
 
+    private final LogsdbIndexModeSettingsProvider logsdbIndexModeSettingsProvider;
```

Comment:
I think this needs to be returned by: `getAdditionalIndexSettingProviders()` method in this class?

Reply:
It is right?

```
return List.of(new SyntheticSourceIndexSettingsProvider(licenseService), logsdbIndexModeSettingsProvider);
```